File: 1603-running-sum-of-1d-array/1603-running-sum-of-1d-array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var runningSum = function(nums) {
    let runningSumArray = [];
    let sum = 0;

    for (let num of nums) {
        sum += num;
        runningSumArray.push(sum);
    }

    return runningSumArray;
};

File: 1603-running-sum-of-1d-array/1603-running-sum-of-1d-array.swift
class Solution {
    func runningSum(_ nums: [Int]) -> [Int] {
        var runningSumArray = [Int]()
        var sum = 0
    
    for num in nums {
        sum += num
        runningSumArray.append(sum)
    }
    
    return runningSumArray
        
    }
}